ext/standard/url.c

```diff
@@ -261,7 +261,7 @@ PHPAPI php_url *php_url_parse_ex(char const *str, size_t length)
 				memcpy(port_buf, p, (e - p));
 				port_buf[e - p] = '\0';
 				port = ZEND_STRTOL(port_buf, NULL, 10);
-				if (port > 0 && port <= 65535) {
+				if ((port > 0 && port <= 65535) || (port == 0 && *port_buf == '0' && e - p == 1)) {
```

Comment:
Why the additional checks here? If we accept `:001` for other port numbers, I don't think zero should get different treatment than that.

Reply:
The problem is that `port==0` also if there is no digit (e.g. `http://example.com:foo/`). We could drop the `e-p==1` check to be more liberal, though.

Reply:
Hi, I was just checking how the contributing process with PHP works and found this bug quite interesting for a beginner like me.

Just one question, why not just check if the port >= 0  in this case? In the case that the port is ":foo" I believe it's invalid no? Since the RFC says that the port number should be a decimal number

Reply:
I suggest that you try that change (checking for `port >= 0`) and run the test in ext/standard/tests/url/ – there'll be several failures; check the *.diff files. :) (Hint: `ZEND_STROL()` is similar to the `(int)` cast in PHP: both ignore any trailing garbage.

Reply:
Nice I will try to play a bit with the source code, tks man